File: Remove the balls.cpp
class Solution {
  public:
    int finLength(int N, vector<int> color, vector<int> radius) {
        stack<pair<int, int>> st;
        for (int i = 0; i < N; i++) {
            int flag = 1;
            pair<int, int> cur = {color[i], radius[i]};
            while (!st.empty() and st.top() == cur) {
                flag = 0;
                st.pop();
            }
            if (flag == 1) st.push(cur);
        }
        return st.size();
    }
};

File: Remove the balls.py
from typing import List


class Solution:
    def finLength(self, N : int, color : List[int], radius : List[int]) -> int:
        stack=[]#creating a stack
        for i in range(N):
            if not stack:#if no element currently in stack than directly add this element 
                stack.append([color[i],radius[i]])
            else :
                if stack[-1][0]==color[i] and stack[-1][1]==radius[i]:#if color and radius matches last element just pop
                    stack.pop()
                else :
                    stack.append([color[i],radius[i]])#else add this element in stack
        return len(stack)#return the final length of stack
